src/scales/scale.time.js

```diff
@@ -554,7 +554,9 @@ module.exports = function() {
 							datasets[i][j] = timestamp;
 						}
 					} else {
-						timestamps.push.apply(timestamps, labels);
+						for (j = 0; j < labels.length; ++j) {
```

Comment:
cache labels.length:
```js
for (j = 0, jLen = labels.length; j < jLen; ++j) {
```

Reply:
Thanks kurkle, lables.length is cached now.